homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -119,11 +127,17 @@ async def _connect(
         password = user_input.get(CONF_PASSWORD) or ""
 
         def _get_connection() -> Connection:
+            if user_input.get(CONF_VERIFY_SSL):
+                requests_session = None
+            else:
+                requests_session = non_verifying_requests_session(user_input[CONF_URL])
```

Comment:
I suggest only to modify the session if the URL starts with `https://` and verification is disabled.

Reply:
Fair enough, done in 4cb0104b993be3a5adc947bf4e1dee030875b2a0, also in `__init__.py`.